File: The-Food-Works-POS/src/app/interfaces/supplier.ts
export interface Supplier {
  SupplierId? : number;
  SupplierName : string;
  SupplierVatNumber? : string;
  SupplierContactNumber? : string;
  SupplierEmailAddress? : string;
  orderDays? : OrderDay[];
  SupplierOrderDayId? : number;
  SupplierOrderDayDescription? : string;
  OrderMethodName? : string;
  SupplierAddressBuildingNumber? : string;
  SupplierAddressStreetName? : string;
  SupplierAddressCity? : string;
  SupplierAddressId? : number;
  SupplierTypeId? : number;
  OrderMethodId?: number;
  SupplierStatusId?:number;
}
export interface OrderDay {
  SupplierOrderDayId? : number;
  SupplierOrderDayDescription : string;
}

export interface SupplierAddress {
  SupplierAddressId?: number;
  SupplierAddressFull?: string;
  SupplierAddressBuildingNumber?: string;
  SupplierAddressStreetName?: string;
  SupplierAddressCity?: string;
  SupplierAddressSuburb?: string;
  SupplierAddressProvince?: string;
  SupplierAddressCountry?: string;
  SupplierAddressDate?: Date;
  SupplierAddressZip?: string;
  SupplierAddressLat?: number;
  SupplierAddressLng?: number;
}

export interface SupplierCombined {
  supplier: Supplier,
  address?: SupplierAddress
}

export interface SupplierTypes {
  SupplierTypeId: number;
  SupplierTypeName: string;
}

export interface SupplierStatus {
  SupplierStatusId: number;
  SupplierStatusName: string;
}
export interface OrderMethods {
  OrderMethodId: number;
  OrderMethodName: string;
}

export interface SuppAdd {
  SupplierId? : number;
  SupplierName : string;
  SupplierVatNumber? : string;
  SupplierContactNumber? : string;
  SupplierEmailAddress? : string;
  orderDays? : OrderDay[];
  SupplierOrderDayId? : number;
  SupplierOrderDayDescription? : string;
  OrderMethodName? : string;
  SupplierTypeId? : number;
  OrderMethodId?: number;
  SupplierStatusId?:number;

  SupplierAddressId?: number;
  SupplierAddressFull?: string;
  SupplierAddressBuildingNumber?: string;
  SupplierAddressStreetName?: string;
  SupplierAddressCity?: string;
  SupplierAddressSuburb?: string;
  SupplierAddressProvince?: string;
  SupplierAddressCountry?: string;
  SupplierAddressDate?: Date;
  SupplierAddressZip?: string;
  SupplierAddressLat?: number;
  SupplierAddressLng?: number;

}

File: The-Food-Works-WebAPI/Model/Supplier.cs
using System;
using System.Collections.Generic;

#nullable disable

namespace The_Food_Works_WebAPI.Model
{
    public partial class Supplier
    {
        public Supplier()
        {
            SupplierOrderDays = new HashSet<SupplierOrderDay>();
            SupplierOrders = new HashSet<SupplierOrder>();
        }

        public int SupplierId { get; set; }
        public string SupplierName { get; set; }
        public string SupplierVatNumber { get; set; }
        public string SupplierContactNumber { get; set; }
        public string SupplierEmailAddress { get; set; }
        public int SupplierTypeId { get; set; }
        public int SupplierStatusId { get; set; }
        public int OrderMethodId { get; set; }
        public int? SupplierAddressId { get; set; }

        public virtual OrderMethod OrderMethod { get; set; }
        public virtual SupplierStatus SupplierStatus { get; set; }
        public virtual SupplierType SupplierType { get; set; }
        public virtual ICollection<SupplierOrderDay> SupplierOrderDays { get; set; }
        public virtual ICollection<SupplierOrder> SupplierOrders { get; set; }
    }
}
